code-challenges.test.js

```diff
@@ -26,37 +26,171 @@ const numbersArray2 = [24, 27, 30, 33, 36]
+  it("should capitalize all words in the array", () => {
+    const result = capitalizeWords(randomNouns1.concat(randomNouns2));
+    expect(result).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database", "Chopsticks", "Mango"]);
+  });
+});
+
+
+// const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+
+// const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+
+// Good failure
+// FAIL  ./code-challenges.test.js
+// evenThree
+//   ✓ should return a string indicating if the number is divisible by three or not (2 ms)
+// capitalizeArr1
+//   ✕ takes in an array of strings named randomNoms and returns and array with all values capitolized
+// indexVowels
+//   ✓ takes in a string and logs the index of the first vowel
+
+// ● capitalizeArr1 › takes in an array of strings named randomNoms and returns and array with all values capitolized
+
+//   ReferenceError: capitalizeWords is not defined
+
+
+
+
+
 const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
 // b) Create the function that makes the test pass.
 
+// Write the code and see the test pass
+
+// PASS  ./code-challenges.test.js
+// evenThree
+//   ✓ should return a string indicating if the number is divisible by three or not (2 ms)
+// capitalizeWords
+//   ✓ should capitalize all words in the array
+// indexVowels
+//   ✓ takes in a string and logs the index of the first vowel (1 ms)
+const capitalizeWords = (arr) => {
+  const joinedArr = randomNouns1.concat(randomNouns2);
+  const result = joinedArr.map(word => word.charAt(0).toUpperCase() + word.slice(1));
+  return result;
+}
+
+
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // Pseudo code:
+//input: takes in a string 
+// output: logs the index of the first vowel
+// process: First I will need to creat my jest test statement that will create good failure by returning the function as undefined.  I can do this by running the jest test with out having the function written out. To make this test pass I will create a function named indexVowels that takes in an arguement of strings (str) as a parameter.  Next I will declare a array with all the vowels as seperate strings "a", "e", "i", "o", "u".  I will need a for loop to iterate over each string in the array by using the .length method and it will iterate to the next value up one by the i++. I will set up my conditional if statement to return the first indexed vowel by using .includes to check and see.  Finally, my catch all will be to return the -1 index if there are no vowels in any of the variables because if I left it at just return i it would only return the length of the string
```

Comment:
I suggest breaking down your pseudo code to be more "to-the-point" and direct.
Also, I don't think you fully understand how the code is working based on what you wrote here. The code is not iterating through the `vowels` array, but the iteration is actually set to run through the inputted string. In this iteration, using a for loop, you are checking to see if the vowels array contains each letter of the string. The iteration stops once it finds the first match since you are using `return` within a for loop, which will stop iteration once the condition is met.